### PR TITLE
GSYE-771: Adapt SCMManager to read MMR, FIT and coefficent_percentage…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
… from scm_properties

## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=master
**GSY_FRAMEWORK_BRANCH**=bug/GSYE-771

### CODE REVIEW

File: src/gsy_e/gsy_e_core/area_serializer.py [hunk 1]
```diff
@@ -174,33 +183,34 @@ def optional(attr):
...
             # For the SCM only use the CoefficientArea strategy.
-            area = CoefficientArea(
-                name, children, uuid, optional("strategy"), config,
-                coefficient_percentage=description.get("coefficient_percentage", 0.0),
-                market_maker_rate=description.get(
-                    "market_maker_rate",
-                    ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE / 100.),
-                feed_in_tariff=description.get(
-                    "feed_in_tariff", GlobalConfig.FEED_IN_TARIFF / 100.,))
+            area = CoefficientArea(name, children, uuid, optional("strategy"), config)
```
Comment (hannesdiedrich) [inline src/gsy_e/gsy_e_core/area_serializer.py:186]: Actual change

File: src/gsy_e/gsy_e_core/simulation/simulation.py [hunk 2]
```diff
@@ -401,26 +435,50 @@ class CoefficientSimulation(Simulation):
...
+            incremental,
+        )
 
     def update_scm_area_properties(self, area: "CoefficientArea"):
         """Update scm_properties for all non-asset areas."""
-        area.update_fee_properties(self._scm_properties)
+        area.update_area_properties(self._scm_properties)
         for child in area.children:
-            if child.children:
+            if child.children is not None:
```
Comment (hannesdiedrich) [inline src/gsy_e/gsy_e_core/simulation/simulation.py:481]: actual change (enables homes with empty list of children)

File: src/gsy_e/models/area/coefficient_area.py [hunk 3]
```diff
@@ -42,37 +41,34 @@ class CoefficientAreaException(Exception):
...
-    def update_fee_properties(self, properties: Dict) -> None:
-        """Update fee_properties."""
+    def update_area_properties(self, properties: Dict) -> None:
+        """Update area_properties."""
         for property_name, fee_dict in properties.get(self.uuid, {}).items():
-            setattr(self.fee_properties, property_name, fee_dict)
+            setattr(self.area_properties, property_name, fee_dict)
         try:
-            self.fee_properties.validate()
+            self.area_properties.validate()
```
Comment (hannesdiedrich) [inline src/gsy_e/models/area/coefficient_area.py:67]: Actual change

File: src/gsy_e/models/area/coefficient_area.py [hunk 4]
```diff
@@ -95,35 +91,17 @@ def cycle_coefficients_trading(self, current_time_slot: DateTime) -> None:
...
-
-        if key_in_dict_and_not_none(kwargs, "coefficient_percentage"):
-            self.coefficient_percentage = self.validate_coefficient_area_setting(
-                kwargs["coefficient_percentage"], "coefficient_percentage")
-        if key_in_dict_and_not_none(kwargs, "market_maker_rate"):
-            self._market_maker_rate = self.validate_coefficient_area_setting(
-                kwargs["market_maker_rate"], "market_maker_rate")
-        if key_in_dict_and_not_none(kwargs, "feed_in_tariff"):
-            self._feed_in_tariff = self.validate_coefficient_area_setting(
-                kwargs["feed_in_tariff"], "feed_in_tariff")
```
Comment (hannesdiedrich) [inline src/gsy_e/models/area/coefficient_area.py:98]: Actual change

File: src/gsy_e/models/area/coefficient_area.py [hunk 5]
```diff
@@ -162,13 +145,15 @@ def trigger_energy_trades(self, scm_manager: "SCMManager") -> None:
     @property
     def market_maker_rate(self) -> float:
         """Get the market maker rate."""
-        return self._market_maker_rate
+        return self.area_properties.AREA_PROPERTIES["market_maker_rate"]
```
Comment (hannesdiedrich) [inline src/gsy_e/models/area/coefficient_area.py:148]: actual change

File: src/gsy_e/models/area/coefficient_area.py [hunk 6]
```diff
@@ -162,13 +145,15 @@ def trigger_energy_trades(self, scm_manager: "SCMManager") -> None:
...
 
     def _change_home_coefficient_percentage(self, scm_manager: "SCMManager") -> None:
         community_total_energy_need = scm_manager.community_data.energy_need_kWh
         home_energy_need = scm_manager.get_home_energy_need(self.uuid)
         if community_total_energy_need != 0:
-            self.coefficient_percentage = home_energy_need / community_total_energy_need
+            self.area_properties.AREA_PROPERTIES["coefficient_percentage"] = (
+                home_energy_need / community_total_energy_need
+            )
```
Comment (hannesdiedrich) [inline src/gsy_e/models/area/coefficient_area.py:156]: actual change

File: tests/test_area_serializer.py [hunk 7]
```diff
@@ -188,11 +205,8 @@ def test_leaf_deserialization_does_not_deserialize_invalid_args():
 def test_leaf_deserialization_scm():
     ConstSettings.MASettings.MARKET_TYPE = SpotMarketTypeEnum.COEFFICIENTS.value
     recovered = area_from_string(
-        '''{
+        """{
              "name": "house",
-             "coefficient_percentage": 0.4,
-             "feed_in_tariff": 0.8,
-             "market_maker_rate": 0.9,
```
Comment (hannesdiedrich) [inline tests/test_area_serializer.py:195]: actual change

File: tests/test_area_serializer.py [hunk 8]
```diff
@@ -202,44 +216,43 @@ def test_leaf_deserialization_scm():
...
-        ''',
-        _create_config()
+        """,
+        _create_config(),
     )
 
-    assert recovered.coefficient_percentage == 0.4
-    assert recovered._feed_in_tariff == 0.8
-    assert recovered.market_maker_rate == 0.9
-
```
Comment (hannesdiedrich) [inline tests/test_area_serializer.py:212]: actual change

**Review by BigTava — DISMISSED** [review]
LGTM Thanks!

**Pull request merged**